app/code/Magento/InventoryShipping/Model/SourceDeduction/SourceDeductionService.php

```diff
@@ -81,32 +82,37 @@ public function __construct(
      */
     public function execute(SourceDeductionRequestInterface $sourceDeductionRequest): void
     {
-        $sourceItemToSave = [];
+        $itemsToSell = $sourceItems = [];
         $sourceCode = $sourceDeductionRequest->getSourceCode();
         $salesChannel = $sourceDeductionRequest->getSalesChannel();
 
         $stockId = (int)$this->stockResolver->get(
             $salesChannel->getType(),
             $salesChannel->getCode()
         )->getStockId();
-        $itemsToSell = [];
         foreach ($sourceDeductionRequest->getItems() as $item) {
             $itemSku = $item->getSku();
             $qty = $item->getQty();
-            $stockItemConfiguration = $this->getStockItemConfiguration->execute(
-                $itemSku,
-                $stockId
-            );
 
-            if ($stockItemConfiguration === null || !$stockItemConfiguration->isManageStock()) {
-                //Product not assigned to Given Stock or we No need to Manage Stock
+            try {
```

Comment:
Generally, we have a lot of boilerplate code like
```
try {
+            $stockItemConfig = $this->getStockItemConfiguration->execute($sku, $stockId);
+        } catch (NoSuchEntityException $e) {
+            // GetStockItemConfiguration throw NoSuchEntityException when SKU is not assigned to Stock
             return false;
         }
```

Maybe it will be better to return some `$stockItemConfiguration ` with predefined values (like manage_stock=0)
If some `sku` doesn't have `configuration` for  some stock will be sensible use fallback on global configuration